backend/src/BackendOnlyStdLib/LibHttpBaseClient.fs

```diff
@@ -149,54 +159,39 @@ module HttpBaseClient =
+      | :? TaskCanceledException ->
         Telemetry.addTags [ "error", true; "error.msg", "Timeout" ]
-        return Error { code = None; error = "Timeout" }
-      | :? System.ArgumentException as e -> // incorrect protocol, possibly more
-        let message =
-          if e.Message = "Only 'http' and 'https' schemes are allowed. (Parameter 'value')" then
-            "Unsupported protocol"
-          else
-            e.Message
-        Telemetry.addTags [ "error", true; "error.msg", message ]
-        return Error { code = None; error = message }
+        return Error Timeout
+
+      | :? System.ArgumentException as e -> // We know of one specific case indicating Unsupported Protocol
+        if e.Message = "Only 'http' and 'https' schemes are allowed. (Parameter 'value')" then
+          Telemetry.addTags [ "error", true; "error.msg", "Unsupported Protocol" ]
+          return Error(BadUrl "Unsupported Protocol")
+        else
+          Telemetry.addTags [ "error", true; "error.msg", e.Message ]
+          return Error(Other e.Message)
+
       | :? System.UriFormatException ->
         Telemetry.addTags [ "error", true; "error.msg", "Invalid URI" ]
-        return Error { code = None; error = "Invalid URI" }
-      | :? IOException as e -> return Error { code = None; error = e.Message }
+        return Error(BadUrl "Invalid URI")
+
+      | :? IOException as e -> return Error(Other e.Message)
+
       | :? HttpRequestException as e ->
+        // This is a bit of an awkward case. I'm unsure how it fits into our model.
+        // We've made a request, and _potentially_ (according to .NET) have a status
+        // code. That should return some sort of Error - but our Error case type
+        // doesn't have a good slot to include the status code. We could have a new
+        // case of `| ErrorHandlingResponse of statusCode: int` but that feels wrong.
         let code = if e.StatusCode.HasValue then int e.StatusCode.Value else 0
+
         Telemetry.addException [ "error.status_code", code ] e
-        let message = e |> Exception.getMessages |> String.concat " "
-        return Error { code = Some code; error = message }
-    }
 
-  let sendRequest
-    (uri : string)
-    (verb : HttpMethod)
-    (reqBody : byte array)
-    (reqHeaders : List<string * string>)
-    : Ply<Dval> =
-    uply {
-      match! httpCall uri verb reqHeaders reqBody with
-      | Ok response ->
-        let responseHeaders =
-          response.headers
-          |> List.map (fun (k, v) ->
-            DTuple(DStr(String.toLowercase k), DStr(String.toLowercase v), []))
-          |> DList
-
-        return
-          [ ("body", DBytes response.body)
-            ("headers", responseHeaders)
-            ("code", DInt(int64 response.code)) ]
-          |> Dval.obj
-          |> Ok
-          |> DResult
-      | Error err -> return DResult(Error(DStr err.error))
+        return Error(Other(Exception.getMessages e |> String.concat " "))
```

Comment:
The existing internal `HttpClient` does an exception type check for:
`| :? HttpRequestException as e ->`.

This is a case where we can't return an "OK" result. Naturally, the other option is to return an Error case - but since we _have_ a status code in some cases, it feels a bit inappropriate to just "drop" the status code and not return it to the user. What do you think?

Reply:
Maybe `Other` should have a status code?

Reply:
An optional one?